src/binding/fortran/mpif_h/buildiface

```diff
@@ -5289,7 +5289,7 @@ sub specialInitStatement {
 sub HelperForRegister_datarep {
     my $OUTFD = $_[0]; 
 
-    my $proto_params = "void*v1, MPI_Fint*v2, MPI_Fint*v3, void*v4, MPI_Offset*v5, MPI_Fint *v6, MPI_Fint*v7, MPI_Fint *ierr";
+    my $proto_params = "void *v1, MPI_Datatype v2, int v3, void *v4, MPI_Offset v5, void *v6";
```

Comment:
Is `ierr` still needed here?

Reply:
`mpi_conversion_fn_null` is simply a dummy function (for the value of address as a constant) and never actually called. So what the parameters are doesn't really matter. So I simply used one from the C prototype. I guess we could do the double cast to `(void*) ` here as well. I felt that the double cast is a silly trick but I don't seem to have solutions for the other cases.

Reply:
I guess what I mean is that the Fortran prototype for datarep conversion contains an error argument: https://www.mpi-forum.org/docs/mpi-3.1/mpi31-report/node334.htm#Node336. Is that not needed in this case?